src/collectd.conf.pod

```diff
@@ -1474,6 +1474,19 @@ in the un-aggregated (per-CPU, per-state) mode as well.
 When set to B<true>, reports the number of available CPUs.
 Defaults to B<false>.
 
+=item B<ReportGuestState> B<false>|B<true>
+
+When set to B<true>, reports the "guest" and "guest_nice" CPU states.
+Defaults to B<false>.
+
+=item B<SubtractGuestState> B<false>|B<true>
```

Comment:
Having two options to control this behavior feels a bit like premature optimization. In other words, are users actually going to use those options? (This is an honest question, not a rhetorical one; I know nothing about the new guest counters ;)

I'd prefer to implement the behavior that `ReportGuestState true; SubtractGuestState true` has and implement config options only if (and when) users ask for it.

Reply:
I have implemented these options to address the following concern:
> But if we start to reduce user and nice, that will be backward-incompatible change for people who do not ready for a such. May be that should be tunable?

I believe having these options keep you safe from any angry user, as they enable end users to revert to whatever old behaviour they were used to. So I would argue that, yes, indeed, some people will use them. Specifically, these counters are notoriously useless within VMs: people with 10 000+ VMs at $RandomCloudCompany may not want to store 20 000+ extra useless metrics. I may even use ReportGuestState myself, depending on the host I deploy collectd to :)

However, we surely can change the **default values**. The actual question is: what is the policy of the collectd project regarding new metrics within existing plugins?

  - Conservative? ``ReportGuestState false; SubtractGuestState true``
  - Progressive? ``ReportGuestState true; SubtractGuestState true``
  - Prone to delegate computations to query and graphing tools? ``ReportGuestState true; SubtractGuestState false``

The choice is yours :)

Reply:
*Conservative:* New versions need to be backwards compatible (with the exception of new major versions). Backwards compatibility is defined by the user observable behavior, i.e. the previous config file needs to still work and all metrics that were reported previously need to still be reported.

I'd consider this a bit of an edge case, sine the previous metrics are still *present*. But it sounds like having users make a conscious decision to enable these metrics is reasonable.

You've made a good case for `ReportGuestState` – who would use `SubtractGuestState`?

Reply:
> who would use SubtractGuestState?
End users who only want the raw values straight out from the kernel to graph them the way they want in one or several graphs. Typically, ``SubtractGuestState true`` is useful for people who want to **stack** all CPU metrics in a single graph and be done with it. But some prefer to mix stacked and non-stacked values on a single graph. Some prefer a separate graph to reflect the host vs guests CPU consumption. 

I tend to focus on graph aficionados but alerting is to be considered too: some alerts and thresholds may assume that *guest* is included in *user*, others may assume that *user* reflects only the host-specific activity.

That being said, since you aim for a conservative policy, what will most likely happen is that end users will notice the lack of *guest* and *guest_nice* (like I did), have a look at the documentation, be glad to find they can enable them easily and hopefully and learn at the same time that subtlety about them being part of *user* and *nice (I didn't know).

Reply:
SGTM, let's go with both options.